### ISSUE TITLE
[PATCH] v5.8.8 pod2html -- option --quota-style us for "q" => ``q'' conversion

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8818]
Migrated from [rt.perl.org#41686](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=41686) (status was 'resolved')

Searchable as RT41686$

**Comment by p5pRT** [comment]
### From @jaalto

#### Created by jaalto@cante.cante.net



The quotation style

&nbsp;    "quotation"

is I believe read and understood all over the world\. However\,
the pod2html converts these into&#8203;:

&nbsp;    \`\`quotation''

which I believe is US specific quotation style \(not used in Europe\)\.

SOLUTION

The default quotation style should stay as it is written in the POD\,
so following patch adds a separate command line option&#8203;:

&nbsp;    \-\-quote\-style\-us

to instruct to specifucally use following conversion for HTML output&#8203;:

&nbsp;    "quotation"  => \`\`quotation''

Jari

Please note\, that this patch is based on previous work that I have
submitted \(see bug report titled&#8203;: "\[perl \#41685\] \[PATCH\] v5\.8\.8
pod2html \-\- Add new option \-\-prelink to treat \<pre>\.\.\</pre> URLs"\)

\+
 =item recurse

&nbsp;     \-\-recurse
@&#8203;@&#8203; \-228\,7 \+234\,7 @&#8203;@&#8203;
 my\($Dircache\, $Itemcache\);
 my @&#8203;Begin\_Stack;
 my @&#8203;Libpods;
\-my\($Htmlroot\, $Htmldir\, $Htmlfile\, $Htmlfileurl\, $Prelink\);
\+my\($Htmlroot\, $Htmldir\, $Htmlfile\, $Htmlfileurl\, $Prelink\, $QuoteStyleUS\);
 my\($Podfile\, @&#8203;Podpath\, $Podroot\);
 my $Css;

@&#8203;@&#8203; \-295\,6 \+301\,7 @&#8203;@&#8203;
&nbsp;     $Verbose = 0;              \# not verbose by default
&nbsp;     $Doindex = 1;              \# non\-zero if we should generate an index
&nbsp;     $Prelink = 0;              \# Treat \<pre>\.\.\</pre> for URLsx
\+    $QuoteStyleUS = 0;         \# Treat "quote" as \`\`quote'' in HTML
&nbsp;     $Backlink = '';            \# text for "back to top" links
&nbsp;     $Listlevel = 0;            \# current list depth
&nbsp;     @&#8203;Listend = \(\);             \# the text to use to end the list\.
@&#8203;@&#8203; \-647\,6 \+654\,8 @&#8203;@&#8203;
&nbsp;   \-\-podroot      \- filesystem base directory from which all relative paths
&nbsp;                    in podpath stem \(default is \.\)\.
&nbsp;   \-\-\[no\]quiet    \- suppress some benign warning messages \(default is off\)\.
\+  \-\-quote\-style\-us
\+                 \- convert "Quotation" into \`\`Quotation'' in HTML
&nbsp;   \-\-\[no\]recurse  \- recurse on those subdirectories listed in podpath
&nbsp;                    \(default behaviour\)\.
&nbsp;   \-\-title        \- title that will appear in resulting html file\.
@&#8203;@&#8203; \-661\,7 \+670\,7 @&#8203;@&#8203;
&nbsp;     my \($opt\_backlink\,$opt\_cachedir\,$opt\_css\,$opt\_flush\,$opt\_header\,$opt\_help\,
&nbsp;        $opt\_htmldir\,$opt\_htmlroot\,$opt\_index\,$opt\_infile\,$opt\_libpods\,
&nbsp;        $opt\_netscape\,$opt\_outfile\,$opt\_prelink\,
\-        $opt\_podpath\,$opt\_podroot\,$opt\_quiet\,
\+        $opt\_podpath\,$opt\_podroot\,$opt\_quote\_style\_us\,$opt\_quiet\,
&nbsp;        $opt\_recurse\,$opt\_title\,$opt\_verbose\,$opt\_hiddendirs\);

&nbsp;     unshift @&#8203;ARGV\, split ' '\, $Config\{pod2html\} if $Config\{pod2html\};
@&#8203;@&#8203; \-683\,6 \+692\,7 @&#8203;@&#8203;
&nbsp;                            'prelink'    => \\$opt\_prelink\,
&nbsp;                            'podpath=s'  => \\$opt\_podpath\,
&nbsp;                            'podroot=s'  => \\$opt\_podroot\,
\+                           'quote\-style\-us' => \\$opt\_quote\_style\_us\,
&nbsp;                            'quiet\!'     => \\$opt\_quiet\,
&nbsp;                            'recurse\!'   => \\$opt\_recurse\,
&nbsp;                            'title=s'    => \\$opt\_title\,
@&#8203;@&#8203; \-708\,6 \+718\,7 @&#8203;@&#8203;
&nbsp;     $Htmlfile = $opt\_outfile  if defined $opt\_outfile;
&nbsp;     $Prelink  = $opt\_prelink  if defined $opt\_prelink;
&nbsp;     $Podroot  = $opt\_podroot  if defined $opt\_podroot;
\+    $QuoteStyleUS = $opt\_quote\_style\_us   if defined $opt\_quote\_style\_us;
&nbsp;     $Quiet    = $opt\_quiet    if defined $opt\_quiet;
&nbsp;     $Recurse  = $opt\_recurse  if defined $opt\_recurse;
&nbsp;     $Title    = $opt\_title    if defined $opt\_title;
@&#8203;@&#8203; \-1439\,8 \+1450\,12 @&#8203;@&#8203;
&nbsp;     if\( $$quote && $text =~ s/"/''/s \)\{
&nbsp;         $$quote = 0;
&nbsp;     \}
\-    while \($text =~ s/"\(\[^"\]\*\)"/\`\`$1''/sg\) \{\};
\-    $$quote = 1 if $text =~ s/"/\`\`/s;
\+
\+    if \( $QuoteStyleUS \)
\+    \{
\+        while \($text =~ s/"\(\[^"\]\*\)"/\`\`$1''/sg\) \{\};
\+        $$quote = 1 if $text =~ s/"/\`\`/s;
\+    \}

&nbsp;     \# keep track of leading and trailing white\-space
&nbsp;     $lead  = \($text =~ s/\\A\(\\s\+\)//s ? $1 : ""\);

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.8.8:

Configured by Debian Project at Wed Dec  6 23:17:41 UTC 2006.

Summary of my perl5 (revision 5 version 8 subversion 8) configuration:
  Platform:
    osname=linux, osvers=2.6.18.3, archname=i486-linux-gnu-thread-multi
    uname='linux saens 2.6.18.3 #1 smp sat nov 25 13:39:52 est 2006 i686 gnulinux '
    config_args='-Dusethreads -Duselargefiles -Dccflags=-DDEBIAN -Dcccdlflags=-fPIC -Darchname=i486-linux-gnu -Dprefix=/usr -Dprivlib=/usr/share/perl/5.8 -Darchlib=/usr/lib/perl/5.8 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/perl5 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.8.8 -Dsitearch=/usr/local/lib/perl/5.8.8 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dsiteman1dir=/usr/local/man/man1 -Dsiteman3dir=/usr/local/man/man3 -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Uusesfio -Uusenm -Duseshrplib -Dlibperl=libperl.so.5.8.8 -Dd_dosuid -des'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -DDEBIAN -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -DDEBIAN -fno-strict-aliasing -pipe -I/usr/local/include'
    ccversion='', gccversion='4.1.2 20061115 (prerelease) (Debian 4.1.1-20)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lgdbm -lgdbm_compat -ldb -ldl -lm -lpthread -lc -lcrypt
    perllibs=-ldl -lm -lpthread -lc -lcrypt
    libc=/lib/libc-2.3.6.so, so=so, useshrplib=true, libperl=libperl.so.5.8.8
    gnulibc_version='2.3.6'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.8:
    /home/jaalto/var/lib/code/perl
    /etc/perl
    /usr/local/lib/perl/5.8.8
    /usr/local/share/perl/5.8.8
    /usr/lib/perl5
    /usr/share/perl5
    /usr/lib/perl/5.8
    /usr/share/perl/5.8
    /usr/local/lib/site_perl
    /usr/local/lib/perl/5.8.7
    /usr/local/share/perl/5.8.7
    /usr/local/lib/perl/5.8.4
    /usr/local/share/perl/5.8.4
    .


Environment for perl v5.8.8:
    HOME=/home/jaalto
    LANG (unset)
    LANGUAGE (unset)
    LC_ALL=en_US
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/bin:/home/jaalto/var/link/bin:/sbin:/bin:/usr/bin:/usr/sbin:/usr/share/bin:/usr/bin/X11:/usr/games
    PERL5LIB=/home/jaalto/var/lib/code/perl
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @jandubois

On Sat\, 03 Mar 2007\, Jari Aalto \(via RT\) wrote&#8203;:
> \[Please enter your report here\]
> 
> The quotation style
> 
>     "quotation"
> 
> is I believe read and understood all over the world\. However\,
> the pod2html converts these into&#8203;:
> 
>     \`\`quotation''
> 
> which I believe is US specific quotation style \(not used in Europe\)\.

This has already been fixed in bleadperl&#8203;:

Applied patch is here&#8203;:

&nbsp;    http&#8203;://public\.activestate\.com/cgi\-bin/perlbrowse/p/28305

Discussion started here&#8203;:

&nbsp;    http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/2006/05/msg113207\.html

Cheers\,
\-Jan

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @rgarcia

On 04/03/07\, Jari Aalto \<jari\.aalto@&#8203;cante\.net> wrote&#8203;:
> Is the patch inside latest Perl yet? Is there 5\.x development code
> accessible through some version control system?

Yes\, see perldoc perlhack too see how it's managed \(bleadperl vs
maintperl\) and how to rsync the latest dev tree\.

**Comment by p5pRT** [comment]
### From @rgs

Marking as solved\, since the problem was already patched in bleadperl\.

**Comment by p5pRT** [comment]
@rgs - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @jaalto

jand@&#8203;activestate\.com \("Jan Dubois"\) writes&#8203;:

> On Sat\, 03 Mar 2007\, Jari Aalto \(via RT\) wrote&#8203;:
>> \[Please enter your report here\]
>> 
>> The quotation style
>> 
>>     "quotation"
>> 
>> is I believe read and understood all over the world\. However\,
>> the pod2html converts these into&#8203;:
>> 
>>     \`\`quotation''
>> 
>> which I believe is US specific quotation style \(not used in Europe\)\.
>
> This has already been fixed in bleadperl&#8203;:
>
> Applied patch is here&#8203;:
>
>     http&#8203;://public\.activestate\.com/cgi\-bin/perlbrowse/p/28305
>
> Discussion started here&#8203;:
>
>     http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/2006/05/msg113207\.html

Is the patch inside latest Perl yet? Is there 5\.x development code
accessible through some version control system?

Jari
